src/main/java/seedu/address/model/InfoHandler.java

```diff
@@ -230,24 +253,47 @@ public VisitBook generateVisitsByPerson(Id personId) {
     }
 
     /**
-     * Generates a list of visits that are associated with the given list of location Ids.
-     * @param locationIds List of location Ids.
-     * @return List of visits that are associated with the location Ids.
+     * Generates a list of visits that are associated with the given visits. This is an intermediate method for the
+     * generatePeopleCommand. After the visits of an infected person have been found, this method finds all other
+     * visits by other people that are associated with the infected person's visits. A visit is considered associated
+     * if the location and date of visit are the same.
+     * @param visitBook Visits provided.
+     * @return List of visits that are associated with the given visits.
      */
-    public VisitBook generateVisitsByLocationIds(List<Id> locationIds) {
+    public VisitBook generateAssociatedVisits(VisitBook visitBook) {
         ReadOnlyVisitBook tempVisitBook = model.getVisitBook();
         VisitBook associatedVisits = new VisitBook();
-        for (Id locationId : locationIds) {
+        for (Visit givenVisit : visitBook.getVisitList()) {
             for (int i = 0; i < tempVisitBook.getVisitList().size(); i++) {
                 Visit visit = tempVisitBook.getVisitList().get(i);
-                if (visit.getLocation().getId().equals(locationId)) {
+                if (isAddableVisit(visit, givenVisit, associatedVisits)) {
                     associatedVisits.addVisit(visit);
                 }
             }
         }
         return associatedVisits;
     }
 
+    /**
+     * Checks if another visit can be added to the list of visits that is associated with the a specified visit.
+     * The visit is considered addable if it meets 3 criteria.
+     * 1. The location of the visit is the same as the specified visit's.
+     * 2. The date of the visit is the same as the specified visit's.
+     * 3. The visit is not already included in the list of visits (This is to prevent duplicates).
+     * This method is only used inside generateAssociatedVisits and it was to prevent conditional statements from
+     * becoming too deeply nested.
+     * @param visit Visit to be checked.
+     * @param givenVisit Visit the above parameter is being checked against.
+     * @param associatedVisits List of visits already affected by the given visit.
+     * @return A boolean of whether the visit can be added or not.
+     */
+    private boolean isAddableVisit(Visit visit, Visit givenVisit, VisitBook associatedVisits) {
```

Comment:
I find that Addable as a method name feels very general. Perhaps addableVisitBy.... with some explanation at the back would be better.

Reply:
Renamed! Hope they all are clearer now.